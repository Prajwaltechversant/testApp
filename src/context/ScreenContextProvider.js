import { useWindowDimensions, Dimensions } from 'react-native'
import React, { useContext, useEffect, useState } from 'react'
// import { isTablet } from 'react-native-device-info';

const ScreenContext = React.createContext()

export const useScreenContext = () => useContext(ScreenContext);

export default function ScreenContextProvider({ children }) {
    const dimensions = useWindowDimensions()

    // screen dimentions
    const width = Dimensions.get('window').width;
    const height = Dimensions.get('window').height;
    const scale = Dimensions.get('window').scale;
    const fontScale = Dimensions.get('window').fontScale;

    // portrait / land

    const isPortrait = height > width;

    // let isTabletType = isTablet()

    // states for dimentions
    const [windowWidth, setWindowWidth] = useState(width);
    const [windowHeight, setWindowHeight] = useState(height);
    const [windowScale, setWindowscale] = useState(scale);
    const [windowFontScale, setWindowFontScale] = useState(fontScale);
    const [windowIsPortrait, setWindowIsPortrait] = useState(isPortrait)


    function setDimensions() {
        const { height, width, scale, fontScale } = dimensions;
        const modPortrait = height > width;

        setWindowHeight(height)
        setWindowWidth(width);
        setWindowFontScale(fontScale)
        setWindowscale(scale)
        setWindowIsPortrait(modPortrait)
    }

    useEffect(() => {
        setDimensions()
    }, [dimensions])

    return (
        <ScreenContext.Provider
            value={{ windowHeight, windowWidth, windowFontScale, windowScale, windowIsPortrait,
            //  isTabletType 
             }}>
            {children}
        </ScreenContext.Provider >

    )
}