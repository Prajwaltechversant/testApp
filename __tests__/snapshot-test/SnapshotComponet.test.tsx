import React from 'react';
import renderer from 'react-test-renderer';
import SnapshotComponet  from '../../src/testing-components/snapshot/index'
import {render}  from '@testing-library/react-native'
import ScreenContextProvider from '../../src/context/ScreenContextProvider';

test('renders correctly', () => {
    const tree = render(

    <ScreenContextProvider><SnapshotComponet  /></ScreenContextProvider>

  ).toJSON();
    expect(tree).toMatchSnapshot();
  });
