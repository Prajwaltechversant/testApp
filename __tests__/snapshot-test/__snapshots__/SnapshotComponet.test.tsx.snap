// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "red",
        "fontSize": 30,
      }
    }
  >
    truefalse
  </Text>
</View>
`;
